pkg/aws/ec2_test.go

```diff
@@ -730,3 +731,235 @@ func TestGetEC2SecurityGroupByNameE(t *testing.T) {
+	encrypted := true
+	volumeType := types.VolumeTypeGp3
+	volumeIops := aws.Int32(100)
+	volumeThroughput := aws.Int32(1000)
+	instanceOutput := &ec2.DescribeInstancesOutput{
+		Reservations: []types.Reservation{
+			{
+				Instances: []types.Instance{
+					{
+						InstanceId: &instanceID,
+						BlockDeviceMappings: []types.InstanceBlockDeviceMapping{
+							{
+								DeviceName: &deviceName,
+								Ebs: &types.EbsInstanceBlockDevice{
+									VolumeId: &volumeId,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	volumeOutput := &ec2.DescribeVolumesOutput{
+		Volumes: []types.Volume{
+			{
+				Encrypted:  &encrypted,
+				KmsKeyId:   &kmsKeyID,
+				VolumeType: volumeType,
+			},
+		},
+	}
+	clientMock := &EC2ClientMock{
+		DescribeInstancesOutput: instanceOutput,
+		DescribeVolumesOutput:   volumeOutput,
+	}
+	fakeTest := &testing.T{}
+
+	// Execute
+	AssertEC2VolumeType(fakeTest, context.Background(), clientMock, AssertVolumeAttributesInput{
+		DeviceID:         deviceName,
+		InstanceID:       instanceID,
+		VolumeType:       "gp3",
+		VolumeIOPS:       volumeIops,
+		VolumeThroughput: volumeThroughput,
+	})
+
+	// Assert
+	assert.False(t, fakeTest.Failed())
+}
+
+func TestAssertEC2VolumeType_MatchWiththroughput(t *testing.T) {
+	// Setup
+	instanceID := "i546acas321sd"
+	volumeId := "v123dfasd92"
+	deviceName := "/dev/sdc"
+	kmsKeyID := "/key/id"
+	encrypted := true
+	volumeType := types.VolumeTypeGp3
+	volumeIops := aws.Int32(100)
+	volumeThroughput := aws.Int32(1000)
+	instanceOutput := &ec2.DescribeInstancesOutput{
+		Reservations: []types.Reservation{
+			{
+				Instances: []types.Instance{
+					{
+						InstanceId: &instanceID,
+						BlockDeviceMappings: []types.InstanceBlockDeviceMapping{
+							{
+								DeviceName: &deviceName,
+								Ebs: &types.EbsInstanceBlockDevice{
+									VolumeId: &volumeId,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	volumeOutput := &ec2.DescribeVolumesOutput{
+		Volumes: []types.Volume{
+			{
+				Encrypted:  &encrypted,
+				KmsKeyId:   &kmsKeyID,
+				VolumeType: volumeType,
+				Throughput: volumeThroughput,
+				Iops:       volumeIops,
+			},
+		},
+	}
+	clientMock := &EC2ClientMock{
+		DescribeInstancesOutput: instanceOutput,
+		DescribeVolumesOutput:   volumeOutput,
+	}
+	fakeTest := &testing.T{}
+
+	// Execute
+	AssertEC2VolumeThroughput(fakeTest, context.Background(), clientMock, AssertVolumeAttributesInput{
+		DeviceID:         deviceName,
+		InstanceID:       instanceID,
+		VolumeType:       "gp3",
+		VolumeIOPS:       volumeIops,
+		VolumeThroughput: volumeThroughput,
+	})
+
+	// Assert
+	assert.False(t, fakeTest.Failed())
+}
+
+func TestAssertEC2VolumeType_MatchWithIops(t *testing.T) {
+	// Setup
+	instanceID := "i546acas321sd"
+	volumeId := "v123dfasd92"
+	deviceName := "/dev/sdc"
+	kmsKeyID := "/key/id"
+	encrypted := true
+	volumeType := types.VolumeTypeGp3
+	volumeIops := aws.Int32(100)
+	volumeThroughput := aws.Int32(1000)
```

Comment:
**Suggestion:** Would you please use the [SDK v2 version of this method](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2@v1.13.0/aws#Int32)?

Reply:
addressed